File: code-coaches/medium/days-between-dates/DaysBetweenDates.java
import java.util.Scanner;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.time.temporal.ChronoUnit;

public class DaysBetweenDates {

    public static int getNumDaysBetween(String fromStr, String toStr) {
        String dateFormat = "MMMM d, yyyy";
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(dateFormat);

        LocalDate fromDate = LocalDate.parse(fromStr, formatter);
        LocalDate toDate = LocalDate.parse(toStr, formatter);

        return (int) ChronoUnit.DAYS.between(fromDate, toDate);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String fromStr = scanner.nextLine();
        String toStr = scanner.nextLine();

        System.out.println(getNumDaysBetween(fromStr, toStr));

        scanner.close();
    }

}

File: code-coaches/medium/days-between-dates/DaysBetweenDates.cs
using System;
using System.Globalization;

public class DaysBetweenDates
{
    static int GetNumDaysBetween(string fromStr, string toStr)
    {
        string dateFormat = "MMMM d, yyyy";
        DateTime fromDate = DateTime.ParseExact(fromStr, dateFormat, CultureInfo.InvariantCulture);
        DateTime toDate = DateTime.ParseExact(toStr, dateFormat, CultureInfo.InvariantCulture);
        TimeSpan span = toDate - fromDate;
        return span.Days;
    }

    static void Main(string[] args)
    {
        string fromStr = Console.ReadLine();
        string toStr = Console.ReadLine();
        Console.WriteLine(GetNumDaysBetween(fromStr, toStr));
    }
}
